lib/Db/AssignmentMapper.php

```diff
@@ -146,4 +146,28 @@ private function getOrigin(Assignment $assignment) {
 		}
 		return null;
 	}
+
+	/**
+	 * @param $ownerId
+	 * @param $newOwnerId
+	 * @return void
+	 */
+	public function transferOwnership($ownerId, $newOwnerId) {
+		$params = [
+			'newOwner' => $newOwnerId,
+			'type' => AssignedUsers::TYPE_USER
+		];
+		$qb = $this->db->getQueryBuilder();
+		$sql = "DELETE FROM `*PREFIX*{$this->tableName}`  WHERE `participant` = :newOwner AND `type`= :type";
```

Comment:
I adapted the logic quite a bit here in a fresh function called remapAssignedUser to process all cards of a given board id. Overall this will end up in having multiple queries per transferred board but processing all user boards that way now also allows to give intermediate status update on the transfer during the running operation.